### PR TITLE
 ZEP-2171: Move all board pinmux code from drivers/pinmux/stm32 to boards/arm/..

### PR DESCRIPTION
**Pull request by erwango**
Move stm32 soc/board specific pinmux files to the corresponding board and SoC directories.
For this, a new "common" folder has been created under arch/arm/soc/st_stm32 in order to gather stm32 common files for various stm32 series.
Following this, pinmux_stm32.c is moved under /drivers/pinmux/

JIRA:  ZEP-2171

### CODE REVIEW

File: arch/arm/soc/st_stm32/common/stm32_pinmux.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (galak) [inline arch/arm/soc/st_stm32/common/stm32_pinmux.h]: I'd rather this live in include/drivers/pinux/stm32.h

File: arch/arm/soc/st_stm32/stm32l4/Makefile [hunk 2]
```diff
@@ -1,3 +1,5 @@
+ZEPHYRINCLUDE += -I$(srctree)/arch/arm/soc/st_stm32/common
```
Comment (galak) [inline arch/arm/soc/st_stm32/stm32l4/Makefile]: We shouldn't need this

File: boards/arm/disco_l475_iot1/Makefile [hunk 3]
```diff
@@ -1,2 +1,6 @@
-# No C files (yet)
-obj- += dummy.o
+ccflags-y += -I$(srctree)/include/drivers
+ccflags-y += -I$(srctree)/drivers
+ccflags-y += -I$(srctree)/drivers/pinmux
+ccflags-y += -I$(srctree)/drivers/pinmux/stm32
```
Comment (galak) [inline boards/arm/disco_l475_iot1/Makefile]: Do we really need all these ccflags-y

I'd hope we don't need any of them.

**Review by galak — CHANGES_REQUESTED** [review]
some various include path changes desired.

File: arch/arm/soc/st_stm32/common/stm32_pinmux.h (hunk 1 above)
Comment (galak) [inline arch/arm/soc/st_stm32/common/stm32_pinmux.h]: ignore me on this, where you put it is fine.

**Comment by galak** [comment]
Ok, was thinking about this more.  We need to restructure the defines a bit as we think about this with regards to device tree.  We need the defines pulled out into some headers that aren't intermixed with the code.

**Comment by erwango** [comment]
@galak: about your latest comment: would it be ok to merge this, and then work on a new mecanism working with device tree? I'd like it could be merged quickly so new board porting integrate it ( and I don't have to rebase again and again)
About device tree: I have to work with Linux guys to see what they have (which would be the basis for integration in Zephyr). I know the already have some stmxxx-pinmux.dtsi.

Other proposal is to close this one and put JIRA-2171 on hold with reason "will be solved with migration to device tree".

**Comment by erwango** [comment]
Please don't merge, I've found an issue on F3 and L4.

**Pull request closed without merging**